### ISSUE TITLE
"else" seemingly not parsed correctly

### ISSUE THREAD

**Issue by RBisges** [Beaglefoot/awk-language-server#17]
I have some working awk scripts to convert data files into other formats. They include if-else constructs, which are shown as errors in your addin to vs-code. I tried several changes like adding semicolon, different spaces in different places, and braces. Some way or other it shows problems.
Any idea?

A minimal test case (Test.awk):
```awk
BEGINFILE {
    if( FILENAME == "Test.awk" )
        print "Test"
    else
        print "not Test"
}
```
gawk -f Test.awk Test.awk
GNU Awk 5.1.0, API: 3.0 (GNU MPFR 3.1.5, GNU MP 6.1.2)
Error message in the addin for this is "Syntax error: missing identifier [5,25]". But of course it works as intended and gawk itself doesn't show any errors.

**Comment by RBisges** [comment]
Sorry, I just noticed that in this case a  semicolon after print will make the error disappear. But the script works fine with or without semicolon

**Comment by Beaglefoot** [comment]
I can not replicate this with version 0.6.2. What version do you have?

**Comment by RBisges** [comment]
Same version. Trying to construct another test case which shows it more conclusively

**Comment by RBisges** [comment]
This here is from one of the scripts. It shows the same error, but at [7,6]
```awk
BEGINFILE {
    if( FILENAME ~ /\.csv$/ )
	{
		FS = ";";
		nCSV++;
	}
	else
	{
		FS = " ";
	}
}
```

**Comment by Beaglefoot** [comment]
The second example also looks fine on my side.

I assume that you're on linux. Could you exec the following 3 commands and paste their output here?

1. `code --list-extensions --show-versions`
2. `~/.vscode/extensions/beaglefoot.awk-ide-vscode-?.?.?/node_modules/awk-language-server/out/cli.js --version`
3. `md5sum ~/.vscode/extensions/beaglefoot.awk-ide-vscode-?.?.?/node_modules/awk-language-server/tree-sitter-awk.wasm`

**Comment by RBisges** [comment]
Sorry, was in a meeting. Unfortunately I have to work on a windows machine for this. It seems that code on windows doesn't have the same cli. I tried to get the inormation in other ways.
```
VS-Code Version:
Version: 1.63.2 (user setup)
Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
Date: 2021-12-15T09:40:02.816Z (2 wks ago)
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19043
``` 
I found your addin under \<USER\>\\\.vscode. I copied that from the manifest:
```
<Identity Language="en-US" Id="awk-ide-vscode" Version="0.6.2" Publisher="beaglefoot" />
```
I created the md5sum with Total Commander.
```
bc892463771af140d3e197bf99ac821d *tree-sitter-awk.wasm
```
I looked into package.json  and it shows me 0.6.2 .

Still missing is the list of extensions and their versions. Searching for a way. But maybe it's just the OS which makes the difference here

**Comment by Beaglefoot** [comment]
Your versions and md5 sum are ok. 

I checked examples above on box with windows both with mine and yours set of extensions and all is fine. Still cannot reproduce.

At this point I suspect that there might be several versions of vscode on your system. What if you run vscode from command line with just `code.cmd` you used earlier? Does the problem persists this way?

**Comment by Beaglefoot** [comment]
Ok, I finally reproduced that and the reason is that on windows lines end with CRLF by default.

Here's quick fix:

- Press Shift+Ctrl+P
- Enter 'Change End of Line Sequence'
- Select 'LF'

**Comment by RBisges** [comment]
I tried on my development notebook now. Different setup, same Windows version (21H1), same VS-Code version.
The effect is the same. I tried starting VS-Code via the code.cmd with no recognisable difference. 
Another strange effect: If I change the position of the opening brace to be on the same line after "else", the warning is gone. But the colourization hints to the parser not recognising "else" as a syntax element.
```awk
BEGIN {
	nCSV = 0;
}

BEGINFILE {
    if( FILENAME ~ /\.csv$/ )
	{
		FS = ";";
		nCSV++;
	}
	else	{
		FS = " ";
	}
}
```
